LibreNMS/OS/Sharp.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/*
+ * Sharp.php
+ *
+ * -Description-
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       https://www.librenms.org
+ * @copyright  2025 Peca Nesovanovic
+ * @author     peca.nesovanovic@sattrakt.com
+ */
+
+namespace LibreNMS\OS;
+
+use App\Models\Device;
+
+class Sharp extends Shared\Printer
+{
+    public function discoverOS(Device $device): void
+    {
+        parent::discoverOS($device); // yaml
+        $device->hardware = $device->hardware ?? \SnmpQuery::get('.1.3.6.1.4.1.2385.2.1.3.2.1.4.13501.1302.1')->value();
```

Comment:
Any reason to do it here instead of YAML for the hardware ?

Reply:
> Any reason to do it here instead of YAML for the hardware ?

hi @PipoCanaja 
main reason to put it here is lack of devices to test
so, this printer will respond to
.1.3.6.1.4.1.2385.2.1.3.2.1.4.13501.1302.1

but ...
i have no other SHARP devices, and test data is very minimal
nobody implemented this before
so i make a future proof idea
$device->hardware ?? $customOids
this way, other SHARP devices could be easy implemented here

Reply:
I don't get it still . Just put whatever you find to identify hardware for your device in the YAML and your are good to go. 
We do that for other OSes where multiple OIDs are tested one ofter the other. Like `includes/definitions/discovery/vrp.yaml` which has this block for hardware mapping : 
```
        hardware:
            - HUAWEI-ENTITY-EXTENT-MIB::hwEntityExtentMIB.6.5.0
            - HUAWEI-DEVICE-EXT-MIB::hwProductName.0
            - HUAWEI-MIB::hwDatacomm.183.1.25.1.5.1
            - HUAWEI-MIB::mlsr.20.1.1.1.3.0
            - HUAWEI-ENTITY-EXTENT-MIB::hwEntityBoardName.9
```

Reply:
ok, done